spark/src/main/scala/geotrellis/spark/tiling/MapKeyTransform.scala

```diff
@@ -122,4 +126,55 @@ class MapKeyTransform(val extent: Extent, val layoutCols: Int, val layoutRows: I
       extent.xmin + (col + 1) * tileWidth,
       extent.ymax - row * tileHeight
     )
+
+  def multiLineToKeys(multiLine: MultiLine): Iterator[SpatialKey] = {
+    val extent = multiLine.envelope
+    val bounds: GridBounds = extentToBounds(extent)
+    val options = Rasterizer.Options(includePartial=true, sampleType=PixelIsArea)
+
+    val boundsExtent: Extent = boundsToExtent(bounds)
+    val rasterExtent = RasterExtent(boundsExtent, bounds.width, bounds.height)
+
+    /*
+     * Use the Rasterizer to construct  a list of tiles which meet
+     * the  query polygon.   That list  of tiles  is stored  as an
+     * array of  tuples which  is then  mapped-over to  produce an
+     * array of KeyBounds.
+     */
+    val tiles = new ConcurrentHashMap[(Int,Int), Unit]
+    val fn = new Callback {
+      def apply(col : Int, row : Int): Unit = {
+        val tile : (Int, Int) = (bounds.colMin + col, bounds.rowMin + row)
+        tiles.put(tile, Unit)
+      }
+    }
+
+    multiLine.foreach(rasterExtent, options)(fn)
+    tiles.keys.asScala.map { case (col, row) => SpatialKey(col, row) }
+  }
+
+  def multiPolygonToKeys(multiPolygon: MultiPolygon): Iterator[SpatialKey] = {
+    val extent = multiPolygon.envelope
+    val bounds: GridBounds = extentToBounds(extent)
+    val options = Rasterizer.Options(includePartial=true, sampleType=PixelIsArea)
+    val boundsExtent: Extent = boundsToExtent(bounds)
+    val rasterExtent = RasterExtent(boundsExtent, bounds.width, bounds.height)
+
+    /*
+     * Use the Rasterizer to construct  a list of tiles which meet
+     * the  query polygon.   That list  of tiles  is stored  as an
+     * array of  tuples which  is then  mapped-over to  produce an
+     * array of KeyBounds.
+     */
+    val tiles = new ConcurrentHashMap[(Int,Int), Unit]
+    val fn = new Callback {
+      def apply(col : Int, row : Int): Unit = {
+        val tile : (Int, Int) = (bounds.colMin + col, bounds.rowMin + row)
+        tiles.put(tile, Unit)
+      }
+    }
```

Comment:
Some dragons been found around these parts, better to use this method: https://github.com/locationtech/geotrellis/blob/5c5ebed09f66cde92c77d9f21ce22e65a44573c4/spark/src/main/scala/geotrellis/spark/tiling/LayoutDefinition.scala#L38

Reply:
Whoa, that's a sexy way to quote the line.

Reply:
@echeipesh I'm using the logic from our LayerFilters for MultiPolygon and MultiLine (which the Line and Polygon filters also use). If this is wrong, then they are also (and have been) wrong. I thought we sorted out all the dragons there - are there some specifics about the dragons?